toxencryptsave/constants.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef TOXENCRYPTSAVE_CONSTANTS_H
+#define TOXENCRYPTSAVE_CONSTANTS_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+// hardcoded constants
+// if they change, userdata will be lost
+
+#define TOXENCRYPTSAVE_MAGIC_NUMBER "toxEsave"
+#define TOXENCRYPTSAVE_MAGIC_LENGTH 8U
+
+#define TOXENCRYPTSAVE_SALT_LENGTH 32U
+#define TOXENCRYPTSAVE_KEY_LENGTH 32U
+#define TOXENCRYPTSAVE_NONCEBYTES 24U
+#define TOXENCRYPTSAVE_ENCRYPTION_EXTRA_LENGTH 80U
+
+namespace toxencryptsave
+{
+    namespace constants
+    {
+        extern const char* magic;
+        static const size_t magic_len = TOXENCRYPTSAVE_MAGIC_LENGTH;
```

Comment:
@ud1 I am trying to rewrite C code into a more readable C++. I think the result code is much better than it used to be. I don't want to spend much time on rewriting the code, so If you want, you may create a PR with your ideas. I have noticed that you are a skilled programmer, so your PR will be likely accepted! 50 tests for `toxencryptsave` make it almost impossible to break the logic. Good luck!